packages/functions/src/jobs/SendConfirmation.test.ts

```diff
@@ -0,0 +1,132 @@
+import * as admin from "firebase-admin";
+import * as Functions from "../index";
+import {
+  AppointmentStatus,
+  BookingChange,
+  FunctionsApi,
+  Hospital,
+} from "@zm-blood-components/common/src";
+import { DbAppointment, DbDonor } from "../function-types";
+import {
+  deleteAppointmentsByIds,
+  getAppointmentsByIds,
+  setAppointment,
+} from "../dal/AppointmentDataAccessLayer";
+import { deleteDonor, setDonor } from "../dal/DonorDataAccessLayer";
+import { sampleUser } from "../testUtils/TestSamples";
+import { mocked } from "ts-jest/utils";
+import {
+  NotificationToDonor,
+  sendEmailToDonor,
+} from "../notifications/NotificationSender";
+import { SendConfirmationReminders } from "./CronJobs";
+import firebaseFunctionsTest from "../testUtils/FirebaseTestUtils";
+
+const wrapped = firebaseFunctionsTest.wrap(
+  Functions[FunctionsApi.BookAppointmentFunctionName]
+);
+
+jest.mock("../notifications/NotificationSender");
+const mockedNotifier = mocked(sendEmailToDonor);
+
+const DONORS = ["BookedDonorId", "AvailableDonorId", "OutOfRangeDonorId"];
+const APPOINTMENTS = [
+  "BookedAppointment",
+  "AvailableAppointment",
+  "OutOfRangeAppointment",
+];
+
+const reset = async () => {
+  for (let i = 0; i < DONORS.length; i++) {
+    await deleteDonor(DONORS[i]);
+  }
+  await deleteAppointmentsByIds(APPOINTMENTS);
+};
+
+beforeAll(async () => {
+  await reset();
+});
+
+afterEach(async () => {
+  await reset();
+});
+
+test("run on all appointments", async () => {
+  await createDonor(DONORS[0]);
+  await createDonor(DONORS[1]);
+  await createDonor(DONORS[2]);
+  const lastHour = new Date();
```

Comment:
I believe test should not use `new Date()`, as this makes them dependent on the running time.

One option to solve this is make `SendConfirmationReminders` also accept `now` date, and set it here in the test to a constant value